### ISSUE TITLE
Deleting a focused item in iron-list causes the focus to return to the first item

### ISSUE THREAD

**Issue by nik3daz** [PolymerElements/iron-list#336]
### Description

Deleting a focused item in iron-list causes the focus to return to the first item.
### Expected outcome

Focus should go to the next item (or previous if the last item is deleted).

### Actual outcome

Focus goes to the first item (possibly even just the first _physical_ item)

### Steps to reproduce
1. Go to https://elements.polymer-project.org/bower_components/iron-list/demo/selection.html
2. Enter `for (var i = 0; i < 200; i++) {$$('x-app')[0].$['itemsList'].selectItem(i)}` in the console
3. Scroll to the bottom of the selected contacts list
4. Click an item
5. Press the Up key.

You can even use keyboard navigation to select items, press enter to delete them and then navigate with the keyboard again to see that the focus goes haywire.

**Comment by blasten** [comment]
That's definitely weird and should be fixed.

**Comment by Parasin** [comment]
Can confirm that I am experiencing the same problem. When using `iron-list` in roughly the same way as described in the problem statement, items appear selected, even though they were never actually selected.

**Comment by nik3daz** [comment]
@Parasin Hey, this seems to be a separate issue. This bug refers to a keyboard focus issue, not a selection issue. Can you please file a separate bug? Thanks.
